### DISCUSSION TITLE
Do we ever work with public key in Aptos contracts?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#14] [Questions] [answered]
An user from [stack overflow asked can we convert get public key from address or signature](https://stackoverflow.com/questions/78024648/can-i-reverse-resolve-the-public-key-from-an-address-or-signature/78110709#78110709).

**Comment by 0x-j** [comment]
I replied on Aptos we never work with user's public key directly, we only work with user's address, so there's no need to support getting public key from address or signature, is that accurate?

**Reply by luongngocminh** [reply]
What if I need to sign an arbitrary message using user A and need a way to verify it on the contract when using user B? Would I have to send both the address and the publickey of the signer to handle the verification?

**Comment by 0x-j** [comment, marked as answer]
More context on relation between address and public key.

The public key is used for verification of transactions, and is found in the transactions directly.  But, outside of that it isn’t necessarily used unless you’re deriving the address from the public key.
Address is a SHA-256 hash from the public key, so there is no way to recover it the other way, unless you basically go to a transaction, with the account, and read the public key from the transaction
